modules/apps/commerce/commerce-inventory-service/src/main/java/com/liferay/commerce/inventory/service/impl/CommerceInventoryBookedQuantityLocalServiceImpl.java

```diff
@@ -228,6 +230,10 @@ public CommerceInventoryBookedQuantity resetCommerceBookedQuantity(
 				commerceInventoryBookedQuantity);
 	}
 
+	@BeanReference(type = CommerceInventoryAuditLocalService.class)
```

Comment:
@slnn Any specific reason to use `@BeanReference` instead of `@ServiceReference` here?

Reply:
Hi @dantewang 

It comes from its base class **CommerceInventoryBookedQuantityLocalServiceImpl**

`@BeanReference(
		type = com.liferay.commerce.inventory.service.CommerceInventoryAuditLocalService.class
	)
	protected
		com.liferay.commerce.inventory.service.
			CommerceInventoryAuditLocalService
				commerceInventoryAuditLocalService;`

Reply:
@slnn OK then, can you try if `@ServiceReference` works?

Reply:
The reference is declared in the base class and used in subclass, after changing into ds, ServiceBuilder will remove them from the base class, here I declare the reference in subclass then change @BeanReference to @Reference after changing to ds, only aim to avoid adding back these references after running build service.

Reply:
@slnn I know. But it seems we always use `@ServiceReference` in `*Impl` even though `@BeanReference` is used in  `*BaseImpl`.

Reply:
Hi @dantewang 

I am trying to apply @ServiceReference to replace @BeanReference in the first commit, but SF complains about this:


`java.lang.Exception: Found 3 formatting issues:
1: Use @BeanReference instead of @ServiceReference: /home/lily/repo/portal-ce/modules/apps/commerce/commerce-inventory-service/src/main/java/com/liferay/commerce/inventory/service/impl/CommerceInventoryBookedQuantityLocalServiceImpl.java 232 (SourceCheck:JavaModuleServiceReferenceCheck)
2: Use @BeanReference instead of @ServiceReference: /home/lily/repo/portal-ce/modules/apps/commerce/commerce-inventory-service/src/main/java/com/liferay/commerce/inventory/service/impl/CommerceInventoryWarehouseItemLocalServiceImpl.java 549 (SourceCheck:JavaModuleServiceReferenceCheck)
3: Use @BeanReference instead of @ServiceReference: /home/lily/repo/portal-ce/modules/apps/commerce/commerce-inventory-service/src/main/java/com/liferay/commerce/inventory/service/impl/CommerceInventoryWarehouseLocalServiceImpl.java 512 (SourceCheck:JavaModuleServiceReferenceCheck)
`


Actually, no matter use @ServiceReference or @BeanReference in the first commit, it will be changed to @Reference later after changing to ds.